python/ray/ml/preprocessors/chain.py

```diff
@@ -13,7 +13,29 @@ class Chain(Preprocessor):
         preprocessors: The preprocessors that should be executed sequentially.
     """
 
-    _is_fittable = False
+    def fit_status(self):
+        fittable_count = 0
+        fitted_count = 0
+        for p in self.preprocessors:
+            # I really cannot see a chain of chain happening. Assert to
+            # explicitly call this out.
+            assert not isinstance(p, Chain)
```

Comment:
I think you could support chain of chains by having a `partially_fitted_count` that you increment when such a partially fitted chain is encountered, then your return stats logic changes a bit:
```python
if fittable_count == 0:
    return Preprocessor.FitStatus.NON_FITTABLE
elif fitted_count == fittable_count:
    return Preprocessor.FitStatus.FITTED
elif fitted_count > 0 or partially_fitted_count > 0:
    return Preprocessor.FitStatus.PARTIALLY_FITTED
else:
    return Preprocessor.FitStatus.NOT_FITTED
```

Reply:
Yeah I think it could potentially work. Though I am a fan of not over generalizing things in favor of clarity and robustness of code. Happy to enhance the logic if we really have a case here.

Reply:
it's fine to start with this.
but probably the comment needs to be updated so it is not from your point of view?? :)

```
# AIR does not support a chain of chained preprocessors at this point.
# Assert to explicitly call this out.
# This can be revisited if compelling use cases emerge.
```